helpers/GenerateResourcesAndImage.ps1

```diff
@@ -240,13 +240,19 @@ Function GenerateResourcesAndImage {
 
     try {
         # Login to Azure subscription
+        try {
```

Comment:
This script will never enter `catch` block because `az account show -o none` command will never throw exception.
Please check other places in this script to see how to handle command output correctly.

Reply:
@maxim-lobanov this is not correct, the `az account show -o none` gives error if not logged in.
Easy to reproduce locally:
```
az logout
az account show -o none
```
`No subscription found. Run 'az account set' to select a subscription.`

See my comment [here](https://github.com/actions/runner-images/pull/10556#issuecomment-2346729920)  why it would never get into the `catch` block